CoconutKit/Sources/ViewControllers/UIViewController+HLSExtensions.m

```diff
@@ -240,23 +240,24 @@ static id swizzled_UIViewController__initWithCoder_Imp(UIViewController *self, S
 
 static void swizzled_UIViewController__viewDidLoad_Imp(UIViewController *self, SEL _cmd)
 {
-    if (! [self isViewLoaded]) {
-        @throw [NSException exceptionWithName:NSInternalInconsistencyException 
-                                       reason:@"The view controller's view has not been loaded" 
-                                     userInfo:nil];
-    }
-    
-    objc_setAssociatedObject(self, s_createdViewSizeKey, [NSValue valueWithCGSize:self.view.bounds.size], OBJC_ASSOCIATION_RETAIN_NONATOMIC);
-    
-    (*s_UIViewController__viewDidLoad_Imp)(self, _cmd);
-    
-    if (! [self isReadyForLifeCyclePhase:HLSViewControllerLifeCyclePhaseViewDidLoad]) {
-        HLSLoggerWarn(@"The viewDidLoad method has been called on %@, but its current view lifecycle state is not compatible. "
-                      "Maybe the view controller is displayed using a container object with incorrect view lifecycle management, "
-                      "or maybe [super viewDidLoad] has not been called by class %@ or one of its parents", self, [self class]);
+    if (![self isViewLoaded]) {
+        HLSLoggerWarn(@"The viewDidLoad method has been called on %@, but its current view has not been loaded yet, we are just going to ignore the call until next call to viewDidLoad has its view loaded", NSStringFromClass([self class]));
+        // The viewDidLoad method has been called on this ViewController, but its current view has not been loaded yet, we are just going to ignore the call until next call to viewDidLoad has its view loaded
+        return;
```

Comment:
instead of throwing an exception we will just ignore the call to "viewDidLoad" until the VC is in correct state so that the VC has a chance to call viewDidLoad again instead of crashing.

This is what happens with 3rd party keyboards in our app, viewDidLoad seems to be called twice for the VC of the keyboard.

Indeed, viewDidLoad seems to be called for the VC of the keyboard while the view has not been prepared yet, then it calls it again when the view has been loaded. The exception is here mostly as a debug tool (like an assert()), this can be useful to catch mistakes during development but in that case it is preventing some code to call viewDidLoad twice.

I am not sure if this is specific to our app or specific to the code of the 3rd party keyboard but having this code will prevent the app from crashing right away when viewDidLoad get called twice and in worst case where viewDidLoad will be called multiples times without still having the view ready and loaded then it will crash too but this will be the responsibility of the creator of that ViewController to do things properly.